ee/session_recordings/session_summary/llm/consume.py

```diff
@@ -0,0 +1,240 @@
+                    summary_prompt=summary_prompt,
+                )
+                if not intermediate_summary:
+                    continue
+                # If parsing succeeds, yield the new chunk
+                sse_event_to_send = _serialize_to_sse_event(
+                    event_label="session-summary-stream", event_data=intermediate_summary
+                )
+                yield sse_event_to_send
+            except SummaryValidationError:
+                # We can except incorrect schemas because of incomplete chunks, ok to skip some.
+                # The stream should be retried only at the very end, when we have all the data.
+                continue
+            except ValueError as err:
+                # The only way to raise ValueError is data hallucinations and inconsistencies (like missing mapping data).
+                # Such exceptions should be retried as early as possible to decrease the latency of the stream.
+                logger.exception(
+                    f"Hallucinated data or inconsistencies in the session summary for session_id {session_id}: {err}"
+                )
+                raise ExceptionToRetry()
+    except (openai.APIError, openai.APITimeoutError, openai.RateLimitError) as err:
+        logger.exception(f"Error streaming LLM for session_id {session_id} by user {user.pk}: {err}")
+        raise ExceptionToRetry()
+    # Final validation of accumulated content (to decide if to retry the whole stream or not)
+    try:
+        if accumulated_usage:
+            TOKENS_IN_PROMPT_HISTOGRAM.observe(accumulated_usage)
+        final_summary = _convert_llm_content_to_session_summary_json(
+            content=accumulated_content,
+            allowed_event_ids=allowed_event_ids,
+            session_id=session_id,
+            simplified_events_mapping=simplified_events_mapping,
+            simplified_events_columns=simplified_events_columns,
+            url_mapping_reversed=url_mapping_reversed,
+            window_mapping_reversed=window_mapping_reversed,
+            session_metadata=session_metadata,
+            summary_prompt=summary_prompt,
+            final_validation=True,
+        )
+        if not final_summary:
+            logger.exception(f"Final LLM content validation failed for session_id {session_id}")
+            raise ValueError("Final content validation failed")
+
+        # If parsing succeeds, yield the final validated summary
+        sse_event_to_send = _serialize_to_sse_event(event_label="session-summary-stream", event_data=final_summary)
+        yield sse_event_to_send
+    # At this stage, when all the chunks are processed, any exception should be retried to ensure valid final content
+    except (SummaryValidationError, ValueError) as err:
+        logger.exception(f"Failed to validate final LLM content for session_id {session_id}: {str(err)}")
+        raise ExceptionToRetry()
+
+
+def _track_session_summary_generation(
```

Comment:
am assuming this is a local testing helper thing?

might be worth commenting it for clarity

(if not, are we sure these paths will save / are useful in prod)